File: code_combat/4_Backwoods_Forest/138-Village_Champion/village_champ.py
def cleaveOrAttack(x, y):
    hero.moveXY(x, y)
    ogre = hero.findNearestEnemy()
    if ogre:
        if hero.isReady("cleave"):
            hero.cleave(ogre)
        else:
            hero.attack(ogre)



while True:
    cleaveOrAttack(35, 34)
    cleaveOrAttack(47, 27)
    cleaveOrAttack(60, 31)

File: projects/code_combat/4_Backwoods_Forest/138-Village_Champion/villageChamp.js
// Incoming munchkins! Defend the town!

// Define your own function to fight the enemy!
function cleaveOrAttack() {
    // In the function, find an enemy, then cleave or attack it.
    var ogre = hero.findNearestEnemy();
    if (ogre) {
        if (hero.isReady("cleave")) {
            hero.cleave(ogre);
        }
        // Else attack the ogre:
        else {
            hero.attack(ogre);
        }
    }
}


// Move between patrol points and call the function.
while (true) {
    hero.moveXY(35, 34);
    // Use cleaveOrAttack function you defined above.
    cleaveOrAttack();
    hero.moveXY(47, 27);
    // Call the function again.
    cleaveOrAttack();
    hero.moveXY(60, 31);
    // Call the function again.
    cleaveOrAttack();
    
}
